src/forms/FormFieldAdvancedSelection/AdvancedSelectionDrawer.tsx

```diff
@@ -64,6 +65,9 @@ const AdvancedSelectionDrawer = (props: AdvanceSelectionDrawerPropTypes): ReactE
 	useEffect(() => {
 		let isMounted = true;
 		if (value?.length > 0 && isModalOpen) {
+			if (value.length >= fieldDef.inputSettings?.selectLimit) {

```

Comment:
I would strongly suggest moving all of these to a function and then just calling it when needed. For example:
```
// my checkboxList disabler function
const disableCheckboxList = useCallback((optionsToCompare) => {
    const selectLimit =  fieldDef?.inputSettings?.selectLimit;  

   if (!selectLimit || selectLimit < 0) return;

    const disableList = selectLimit === 0 || optionsToCompare.length >= selectLimit;
    setCheckboxListDisabled(disableList);

}, [fieldDef?.inputSettings?.selectLimit]);

//Here in this useEffect
useEffect(() => {
    let isMounted = true;
    if (value?.length > 0 && isModalOpen) {
         disableCheckboxList(value);
         ...
    }
}
```